### PR TITLE
[JENKINS-69916] Un-inline `WorkflowJob/configure-entries.jelly`

### PR DESCRIPTION
**Pull request by krisstern**
### Description
**Testing done**. Tried the functionality which appears intact after the code modifications, except for the original testing steps outlined in https://issues.jenkins.io/browse/JENKINS-69916. The checkURL validation is now completely gone. This work follows the upstream changes in https://github.com/jenkinsci/jenkins/pull/8866, especially the ones in `core/src/main/java/hudson/model/AbstractProject.java`. The only requirement for this patch to work is that we will need the minimum Jenkins version requirement to set at `2.443`. 

### JIRA Issue
https://issues.jenkins.io/browse/JENKINS-69916

### Relevant Upstream Change(s)
https://github.com/jenkinsci/jenkins/pull/8866

### Testing Steps
Similar to the ones described in JIRA ticket, except the `checkDisplayName` validation is gone. 

c.c. @Kevin-CB 

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by mawinter69** [comment]
You can't depend on https://github.com/jenkinsci/jenkins/pull/8866 because this change AbstractProject and WorkflowJob is not inheriting from it. So no need to change the jenkins version. You would need to also implement the doCheckDisplayNameOrNull method.

Given that we have this now in MatrixProject (open PR https://github.com/jenkinsci/matrix-project-plugin/pull/130), core (the mentioned change), folder plugin (https://github.com/jenkinsci/cloudbees-folder-plugin/blob/233481b4bb4accfe8582ad2f5c922168022af31d/src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java#L138) and here maybe we can have an AbstractItemDescriptor interface with a default implementation for that doCheckDisplayNameOrNull and have all 4 places implement that interface

**Comment by krisstern** [comment]
Hi @mawinter69 I have tried my best to apply the suggesions from your review. But I might have missed some important details though. If that if the case please let me know. Thanks!

**Comment by mawinter69** [comment]
> Hi @mawinter69 I have tried my best to apply the suggesions from your review. But I might have missed some important details though. If that if the case please let me know. Thanks!

Sorry seems my wording was not precise enough. The thing with the `AbstractItemDescriptor` is something that would need to be implemented in Jenkins core, not in a plugin. This would allow the 4 places that now all have duplicated code for the `doCheckDisplayNameOrNull ` method now to just add the dependency on that interface.

So for now you will need to add this method to the  Descriptor of WorkflowJob, not WorkflowJob directly

**Comment by krisstern** [comment]
Let me follow up on this shortly

**Comment by krisstern** [comment]
Working on the upstream PR pending review(s): https://github.com/jenkinsci/jenkins/pull/9150

File: src/main/java/org/jenkinsci/plugins/workflow/job/AbstractItemDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.jenkinsci.plugins.workflow.job;
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/workflow/job/AbstractItemDescriptor.java:1]: Delete this file.

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java [hunk 2]
```diff
@@ -69,6 +69,7 @@
 import hudson.triggers.TriggerDescriptor;
 import hudson.util.AlternativeUiTextProvider;
 import hudson.util.DescribableList;
+import hudson.util.FormValidation;
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java:72]: Revert, unused.

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java [hunk 3]
```diff
@@ -98,12 +99,14 @@
 import org.kohsuke.accmod.Restricted;
 import org.kohsuke.accmod.restrictions.DoNotUse;
 import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.QueryParameter;
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java:103]: Revert, unused.

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java [hunk 4]
```diff
@@ -98,12 +99,14 @@
...
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.StaplerResponse;
 import org.kohsuke.stapler.export.Exported;
 
 @SuppressWarnings({"unchecked", "rawtypes"})
-public final class WorkflowJob extends Job<WorkflowJob,WorkflowRun> implements LazyBuildMixIn.LazyLoadingJob<WorkflowJob,WorkflowRun>, ParameterizedJobMixIn.ParameterizedJob<WorkflowJob, WorkflowRun>, TopLevelItem, Queue.FlyweightTask, SCMTriggerItem, BlockableResume {
+public final class WorkflowJob extends Job<WorkflowJob,WorkflowRun> implements LazyBuildMixIn.LazyLoadingJob<WorkflowJob,WorkflowRun>, ParameterizedJobMixIn.ParameterizedJob<WorkflowJob, WorkflowRun>, TopLevelItem, Queue.FlyweightTask, SCMTriggerItem, BlockableResume, AbstractItemDescriptor {
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java:109]: Revert.

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java [hunk 5]
```diff
@@ -193,7 +196,6 @@ public void setDefinition(FlowDefinition definition) {
         getTriggersJobProperty().startTriggers(Items.currentlyUpdatingByXml());
     }
 
-
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java:196]: Revert.

**Review by daniel-beck — CHANGES_REQUESTED** [review]
With the core change, should be good enough just deleting the `checkUrl`?

Should increase the core dependency to the snapshot (and then the release once out).

File: src/main/java/org/jenkinsci/plugins/workflow/job/AbstractItemDescriptor.java (hunk 1 above)
Comment (krisstern) [reply src/main/java/org/jenkinsci/plugins/workflow/job/AbstractItemDescriptor.java]: Done

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java (hunk 2 above)
Comment (krisstern) [reply src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java]: Done

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java (hunk 3 above)
Comment (krisstern) [reply src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java]: Done

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java (hunk 4 above)
Comment (krisstern) [reply src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java]: Done

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java (hunk 5 above)
Comment (krisstern) [reply src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java]: Done

**Comment by krisstern** [comment]
Hi @daniel-beck I have just made the changes as suggested

**Review by daniel-beck — APPROVED** [review]
Is this significant enough to need a core dependency with https://github.com/jenkinsci/jenkins/pull/9150, or can we tolerate this not working with older cores?

**Review by basil — APPROVED** [review]
PCT: https://github.com/jenkinsci/bom/pull/3657
ATH: https://github.com/jenkinsci/acceptance-test-harness/pull/1751

**Comment by basil** [comment]
Retested manually against 2.477.x

File: pom.xml [hunk 6]
```diff
@@ -63,21 +63,30 @@
...
+        <jenkins-test-harness.version>2254.vcff7a_d4969e5</jenkins-test-harness.version>
+        <maven.compiler.release>17</maven.compiler.release>
     </properties>
     <dependencyManagement>
         <dependencies>
             <dependency>
                 <groupId>io.jenkins.tools.bom</groupId>
-                <artifactId>bom-2.452.x</artifactId>
-                <version>3023.v02a_987a_b_3ff9</version>
+                <artifactId>bom-2.462.x</artifactId>
```
Comment (jglick) [inline pom.xml:79]: Is a `bom-2.477.x` in preparation?

File: pom.xml [hunk 7]
```diff
@@ -63,21 +63,30 @@
...
+                <version>3221.ve8f7b_fdd149d</version>
                 <scope>import</scope>
                 <type>pom</type>
             </dependency>
+            <!-- TODO JENKINS-73339 until in parent POM, work around https://github.com/jenkinsci/plugin-pom/issues/936 -->
```
Comment (jglick) [inline pom.xml:84]: pending https://github.com/jenkinsci/plugin-pom/pull/1004#issuecomment-2379382600 IIUC

**Pull request merged**